Data/Euclidean.hs

```diff
@@ -147,7 +148,19 @@ infixl 7 `rem`
 coprimeIntegral :: Integral a => a -> a -> Bool
 coprimeIntegral x y = (odd x || odd y) && P.gcd x y == 1
 
--- | A 'Field' represents a
+-- | Execute the extended Euclidean algorithm.
+-- For elements @a@ and @b@, compute their greatest common divisor @g@
+-- and the coefficient @s@ satisfying @as + bt = g@.
+gcdExt :: (Eq a, Euclidean a, Ring a) => a -> a -> (a, a)
+gcdExt = go one zero
+  where
+    go s s' r r'
+      | r' == zero = (r, s)
+      | otherwise  = case quotRem r r' of
+        (q, r'') -> go s' (minus s (times q s')) r' r''
+{-# INLINE gcdExt #-}
```

Comment:
There is not much benefit from inlining a recursive function. However, we would like to specialize it at use sites. Please use `{-# INLINABLE gcdExt #-}`.